src/crystal/system/unix/path.cr

```diff
@@ -11,14 +11,13 @@ module Crystal::System::Path
       pwd_pointer = pointerof(pwd)
       ret = nil
       System.retry_with_buffer("getpwuid_r", User::GETPW_R_SIZE_MAX) do |buf|
-        ret = LibC.getpwuid_r(id, pwd_pointer, buf, buf.size, pointerof(pwd_pointer))
+        ret = LibC.getpwuid_r(id, pwd_pointer, buf, buf.size, pointerof(pwd_pointer)).tap do |ret|
+          # It's not necessary to check success with `ret == 0` because `pwd_pointer` will be NULL on failure
+          return String.new(pwd.pw_dir) if pwd_pointer
+        end
```

Comment:
**question**: why handle `ret` at all? Wouldn't the following be enough:

```crystal
LibC.getpwuid(...)
return String.new(pwd.pw_dir) if pwd_pointer
```

Reply:
The block for `retry_with_buffer` needs to output the return value because it does some error handling.

Reply:
Buf `.tap do |ret|` looks unused, right? Same for the other cases?

Reply:
True, we don't use `ret` (my original implementation did, before I noticed that truthiness of the pointer is sufficient to indicate success.
I'd still keep the `tap`, though to make sure that we're passing the value forward. But the block argument can be removed.